addons/addon-raas-s3-copy/packages/s3-synchronizer/src/s3-upload-file-watcher.go

```diff
@@ -0,0 +1,149 @@
+// Adapted from https://github.com/andymotta/s3-fsnotify-go
+package main
+
+import (
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+	"github.com/fsnotify/fsnotify"
+)
+
+func setupUploadWatcher(sess *session.Session, config *mountConfiguration, debug bool) error {
```

Comment:
I am assuming we are adding unit tests for this later, and not merging this in master right away?

Reply:
Yes. Will add unit tests around this when implementing writes in next PR.